src/serac/physics/materials/parameterized_solid_functional_material.hpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2019-2022, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file solid_functional_material.hpp
+ *
+ * @brief The material and load types for the solid functional physics module
+ */
+
+#pragma once
+
+#include "serac/numerics/functional/functional.hpp"
+#include "serac/physics/materials/solid_functional_material.hpp"
+
+/// SolidFunctional helper data types
+namespace serac::solid_util {
+
+/**
+ * @brief Linear isotropic elasticity material model
+ *
+ * @tparam dim Spatial dimension of the mesh
+ */
+template <int dim>
+class ParameterizedLinearIsotropicSolid {
+public:
+  /**
+   * @brief Construct a new Linear Isotropic Elasticity object
+   *
+   * @param density Density of the material
+   * @param shear_modulus_offset Shear modulus offset of the material
+   * @param bulk_modulus_offset Bulk modulus offset of the material
+   */
+  ParameterizedLinearIsotropicSolid(double density = 1.0, double shear_modulus_offset = 1.0,
+                                    double bulk_modulus_offset = 1.0)
+      : density_(density), bulk_modulus_offset_(bulk_modulus_offset), shear_modulus_offset_(shear_modulus_offset)
+  {
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear isotropic elasticity material model.");
+  }
+
+  /**
+   * @brief Material response call for a linear isotropic solid
+   *
+   * @tparam T1 Spatial position type
+   * @tparam T2 Displacement type
+   * @tparam T3 Displacement gradient type
+   * @tparam T4 Bulk modulus type
+   * @tparam T5 Shear modulus type
+   * @param du_dX Displacement gradient with respect to the reference configuration (du_dX)
+   * @param bulk_parameter The parameterized bulk modulus
+   * @param shear_parameter The parameterized shear modulus
+   * @return The calculated material response (density, kirchoff stress) for the material
+   */
+  template <typename T1, typename T2, typename T3, typename T4, typename T5>
+  SERAC_HOST_DEVICE auto operator()(const T1& /* x */, const T2& /* displacement */, const T3& du_dX,
+                                    const T4& bulk_parameter, const T5& shear_parameter) const
+  {
+    auto bulk_modulus  = bulk_parameter + bulk_modulus_offset_;
+    auto shear_modulus = shear_parameter + shear_modulus_offset_;
+
+    auto I      = Identity<dim>();
+    auto lambda = bulk_modulus - (2.0 / dim) * shear_modulus;
+    auto strain = 0.5 * (du_dX + transpose(du_dX));
+    auto stress = lambda * tr(strain) * I + 2.0 * shear_modulus * strain;
+
+    using StressType = decltype(stress);
+
+    return MaterialResponse<double, StressType>{.density = density_, .stress = stress};
```

Comment:
The `decltype` here looks unpleasant, does
```cpp
return MaterialResponse{.density = density_, .stress = stress};
```
or
```cpp
return MaterialResponse{density_, stress};
```
work just the same? 

If not, I think we might be able to change the implementation of `MaterialResponse` type-- I really don't want the user to be required to manually deduce certain types in their q-functions.

Reply:
Let me check - I had to do a lot of compiler error chasing to get this to work. I'll see if it was something else I needed.

Reply:
If I recall correctly, clang couldn't resolve the class template argument deduction properly. I'll test this though.

Reply:
Well this seems to be working now, so I'll remove the explicit template types.

Reply:
I spoke too soon! The linear version works, but the neo-Hookean model doesn't. I will investigate.

Reply:
if something like
```cpp
return MaterialResponse{.density = density_, .stress = stress};
```
doesn't work, I don't think it should hold up this PR. We can just put a `// TODO` on it and make an issue, to revisit later.